src/main/java/org/jabref/logic/xmp/XmpUtilWriter.java

```diff
@@ -159,6 +171,26 @@ private static void writeDublinCore(PDDocument document,
         catalog.setMetadata(metadataStream);
     }
 
+    public static String generateXmpString(List<BibEntry> entries) {
+        XMPMetadata meta = XMPMetadata.createXMPMetadata();
+        for (BibEntry entry : entries) {
+            DublinCoreSchema dcSchema = meta.createAndAddDublinCoreSchema();
+            XmpUtilWriter.writeToDCSchema(dcSchema, entry, null, Globals.prefs.getXMPPreferences());
+        }
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
```

Comment:
You should always wrap those Input and Output stream in the try with resources construct, that ensures that the streams are autoclosed:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html